webrender_api/src/display_item.rs

```diff
@@ -155,6 +155,13 @@ pub struct StickyFrameDisplayItem {
     /// position is scrolled out of view. Constraints specify a maximum and minimum offset from the
     /// original position relative to non-sticky content within the same scrolling frame.
     pub horizontal_offset_bounds: StickyOffsetBounds,
+
+    /// The amount of offset that has already been applied to the sticky frame. A positive y
+    /// component in this field means that a top-sticky item was in a scrollframe that
+    /// has been scrolled down, such that the sticky item's position needed to be offset
+    /// downwards by `applied_offset.y`. A negative y component corresponds to the upward
+    /// offset applied due to bottom-stickiness. The x-axis works analogously.
+    pub applied_offset: LayoutVector2D,
```

Comment:
Maybe this should be called something like `previously_applied_offset` to avoid confusing it with the current offset provided by WebRender?

Reply:
I was thinking of that too but wasn't sure how about variable name length. I can change it.